tests/test_270_005010X279A1.py

```diff
@@ -108,3 +110,126 @@ def test_property_usage(x12_270_subscriber_input):
         ]
         assert info_source_name["name_last_or_organization_name"] == "PAYER C"
         assert info_source_name["identification_code"] == "12345"
+
+
+def test_loop_and_segment_validations(invalid_270_input):
+    """
+    Validates the loop and segment based validations which are not executed via a root validator.
+    try/except is used, rather than pytest.raises, as we will execute assertions against the expected validations.
+    """
+
+    try:
+        with X12ModelReader(invalid_270_input) as r:
+            for _ in r.models():
+                pass
+    except ValidationError as ve:
+        assert len(ve.raw_errors) == 1
```

Comment:
I will need to go back and update the invalid input to trigger each validation error.
I want to verify that all errors are collected.

Reply:
@ccorley - After reviewing the way validations are configured in this branch, I am going to update things so that we are running Pydantic's "post" validations rather than "pre". 

Running custom validators "post" validation means that Pydantic will validate the schema/field constraints, and then run custom functions. This allows us to "gather up" all of the validation errors within a transaction which imo is more helpful than failing on the first validation. The caveat is that we may have some "noise" mixed into the validations as well since Pydantic does not include "invalid" fields for custom validators in their "values" argument.

Reply:
Makes sense!

Reply:
I have migrated the validators to "post" validation so that we can be consistent across the board. The majority of changes here were in regards to how fields are accessed. In "pre" validation data is accessed via dictionary semantics, since it hasn't Pydantic hasn't loaded it into a model yet - hence the "pre". For the "post" validation  I updated the validations to use "property" syntax such as "nm1_segment.name_first", etc.

I opted to remove the test case above as the only way to validate the number of issues captured is to use a string comparison. This is a bit too brittle to have in a test case, imo.